core/templates/pages/topic-editor-page/modal-templates/topic-editor-save-modal.template.html

```diff
@@ -12,7 +12,7 @@ <h3>
 </div>
 <div class="modal-footer">
   <button class="btn btn-secondary" ng-click="cancel()" type="button">Cancel</button>
-  <button class="btn btn-success protractor-test-close-save-modal-button" ng-disabled="!commitMessage" ng-click="confirm(commitMessage)">
+  <button class="btn btn-success protractor-test-close-save-modal-button" ng-disabled="!commitMessage && isTopicPublished" ng-click="confirm(commitMessage||'commit-before-publishing')">
```

Comment:
Can we just leave commitMessage be an empty string when it is not required?

Reply:
Sure! Done!